staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -44,6 +44,7 @@ var BannedOwners = map[schema.GroupVersionKind]struct{}{
 func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for k := range annotations {
+		// Uppercase is valid for annotation key but not QualifiedName, convert it as lowercase
```

Comment:
nit: IIUC, in particular, uppercase is invalid in a QualifiedName's **prefix**:
- `a.b/C` is a valid QualifiedName
- `A.b/C` is not

The metadata.annotation [doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set) says "the prefix must be a DNS subdomain" which is a little unclear. DNS is case insensitive and can be normalized. While a QualifiedName is expected to have a [lowercase](https://github.com/kubernetes/kubernetes/blob/cae22d8b8a78985f438c232357aa2b6c60d83f9b/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L200) (normalized) RFC 1123 subdomain prefix, today we allow uppercase in metadata.annotation's prefix.